satellite/metabase/rangedloop/rangedlooptest/callbackobserver.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package rangedlooptest
+
+import (
+	"context"
+	"time"
+
+	"storj.io/storj/satellite/metabase/rangedloop"
+	"storj.io/storj/satellite/metabase/segmentloop"
+)
+
+var _ rangedloop.Observer = (*CallbackObserver)(nil)
+var _ rangedloop.Partial = (*CallbackObserver)(nil)
+
+// CallbackObserver needs a comment to pass linting.
```

Comment:
better comments are needed for this file :)

Reply:
I've implemented the rest of the callbacks and added comments.